docs/src/prototypes/customToolbar/CustomToolbar.tsx

```diff
@@ -0,0 +1,341 @@
+    variables: { primary: true },
+    fitted: 'horizontally',
+    content: (
+      <FocusableFlexBox fill gap="gap.medium" vAlign="center">
+        10:45
+      </FocusableFlexBox>
+    ),
+  },
+
+  { key: 'timer-divider', kind: 'divider' },
+
+  {
+    active: props.cameraActive,
+    icon: {
+      name: props.cameraActive ? 'call-video' : 'call-video-off',
+      size: 'large',
+    },
+    key: 'camera',
+    onClick: () => _.invoke(props, 'onCameraChange', !props.cameraActive),
+    variables: { primary: true },
+  },
+
+  {
+    active: props.micActive,
+    icon: {
+      name: props.micActive ? 'mic' : 'mic-off',
+      size: 'large',
+    },
+    key: 'mic',
+    onClick: () => _.invoke(props, 'onMicChange', !props.micActive),
+    variables: { primary: true },
+  },
+
+  {
+    active: props.screenShareActive,
+    icon: {
+      name: props.screenShareActive ? 'call-control-close-tray' : 'call-control-present-new',
+      size: 'large',
+    },
+    key: 'screen-share',
+    onClick: () => _.invoke(props, 'onScreenShareChange', !props.screenShareActive),
+    variables: { primary: true },
+  },
+
+  {
+    key: 'more',
+    icon: {
+      name: 'more',
+      size: 'large',
+    },
+
+    onClick: () => {
+      _.invoke(props, 'onMoreClick')
+    },
+
+    variables: {
+      primary: true,
+    },
+  },
+]
+
+const sidebarButtons: CustomToolbarLayout = props => [
+  {
+    active: props.sidebarSelected === 'chat',
+    icon: {
+      name: 'chat',
+      outline: 'true',
+      size: 'large',
+    },
+    key: 'chat',
+    onClick: () =>
+      _.invoke(props, 'onSidebarChange', props.sidebarSelected === 'chat' ? false : 'chat'),
+    variables: { hasDot: props.chatHasDot, noFillOnHover: true },
+  },
+  {
+    active: props.sidebarSelected === 'participant-add',
+    icon: {
+      name: 'participant-add',
+      outline: 'true',
+      size: 'large',
+    },
+    key: 'participant-add',
+    onClick: () =>
+      _.invoke(
+        props,
+        'onSidebarChange',
+        props.sidebarSelected === 'participant-add' ? false : 'participant-add',
+      ),
+    // TODO: odd to have icon style implementation details leaking here.  works for now, investigate other options.
+    variables: { noFillOnHover: true },
+  },
+]
+
+const layoutItems = {
+  endCall: props => ({
+    key: 'end-call',
+    icon: {
+      name: 'call-end',
+      size: 'large',
+    },
+
+    onClick: () => {
+      _.invoke(props, 'onEndCallClick')
+    },
+
+    variables: {
+      danger: true,
+    },
+  }),
+}
+
+const layouts: Record<CustomToolbarProps['layout'], CustomToolbarLayout> = {
+  standard: props => [...commonLayout(props), ...sidebarButtons(props), layoutItems.endCall(props)],
+
+  'desktop-share': props => [
+    ...commonLayout(props),
+    ...sidebarButtons(props),
+    { key: 'divider-sidebar', kind: 'divider' },
+    {
+      key: 'stop-sharing',
+      kind: 'custom',
+      content: <Button content="Stop Sharing" />,
```

Comment:
we will need to consider the case when this is transformed to a menu item in the overflow menu. It should not be a button there - menu only allows menu items, menu item checkboxes and menu item radios.
Btw this also shows why the custom item type should not have onClick prop available - the user needs to add onClick to the button. Challenge is that for the overflow menu item, the onClick will need to be on the menu item, not on something what is inside of it.

Reply:
> we will need to consider the case when this is transformed to a menu item in the overflow menu. It should not be a button there - menu only allows menu items, menu item checkboxes and menu item radios.

We don't touch responsive layout in this case, but it's a good mention. Should be considered in future updates

> Btw this also shows why the custom item type should not have onClick prop available - the user needs to add onClick to the button.

It was addressed previously with PR for `custom` `kind`